linkerd/app/core/src/accept_error.rs

```diff
@@ -32,12 +37,19 @@ where
     }
 
     fn call(&mut self, sock: T) -> Self::Future {
-        self.0.accept(sock).then(|v| {
-            if let Err(e) = v {
+        self.0.accept(sock).then(|x| match x {
+            Ok(connection_future) => Ok(Box::new(connection_future.then(|v| {
+                if let Err(e) = v {
+                    let error: Error = e.into();
+                    tracing::debug!(%error, "Connection failed");
+                }
+                Ok(())
+            }))),
```

Comment:
Nit, take it or leave it: I think we can get rid of the `if let Err(e) = v` if we change the `then` to `or_else`:
```suggestion
            Ok(connection_future) => Ok(Box::new(connection_future.or_else(|e| {
                let error: Error = e.into();
                tracing::debug!(%error, "Connection failed");
                Ok(())
            }))),
```
`then` closures are called with the result of a future, whether it's `Ok` _or_ `Err`, while `or_else` is called only with an `Err` result but allows returning `Ok` to recover.

Reply:
addressed in https://github.com/linkerd/linkerd2-proxy/pull/471